bids/layout/models.py

```diff
@@ -10,12 +10,16 @@
 from functools import lru_cache
 from collections import UserDict
 
-from sqlalchemy.ext.declarative import declarative_base
 from sqlalchemy.ext.associationproxy import association_proxy
 from sqlalchemy.orm.collections import attribute_mapped_collection
-from sqlalchemy import (Column, String, Boolean, ForeignKey, Table)
+from sqlalchemy import Column, String, Boolean, ForeignKey, Table
 from sqlalchemy.orm import reconstructor, relationship, backref, object_session
 
+try:
+    from sqlalchemy.orm import declarative_base
+except ImportError:  # sqlalchemy < 1.4
+    from sqlalchemy.ext.declarative import declarative_base
```

Comment:
sweet! Thank you @effigies . Since it is somewhat of an explicit "support old and new" - may be worth adding a CI run which would test with `sqlalchemy 1.3.24` ?

or since it has been years since 1.4 release, just require `>= 1.4`?

Reply:
You'll have to wait for a new release of `jupyter_cache` that will restore support of sqlalchemy 1.3. Otherwise handle complex dependencies:
- latest `mist-nb` ⇒ `jupyter_cache` 0.6.0 ⇒`sqlalchemy` ≥ 1.4
- `sqlalchemy` < 1.4⇒previous `nb_mist` or upcoming `mist-nb` that would pull upcoming `jupyter_cache` that would restore support for `sqlalchemy` 1.3.

Reply:
I've generally wanted an explicit test of minimum dependencies, but never got around to it. It seems bad to go from requiring `<1.4` to `>=1.4` in a single commit.

Reply:
I created #987 to add a min-requirements.txt file and test it. It is currently passing locally but failing with what is very likely an out-of-memory error on GHA. How about we continue this conversation there?